web-components/src/components/form/Form.ts

```diff
@@ -0,0 +1,189 @@
+/**
+ * Copyright (c) Cisco Systems, Inc. and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ */
+import { customElement, html, LitElement, property, query } from "lit-element";
+import { ifDefined } from "lit-html/directives/if-defined";
+import { SlottedMixin } from "@/mixins";
+
+export namespace Form {
+  export type Method = "GET" | "POST" | "dialog";
+  export type Enctype = "application/x-www-form-urlencoded" | "multipart/form-data" | "text/plain";
+  export type Target = "_self" | "_blank" | "_parent" | "_top";
+  export type SubmittableElement = HTMLInputElement | HTMLSelectElement | HTMLTextAreaElement;
+
+  @customElement("md-form")
+  export class ELEMENT extends SlottedMixin(LitElement) {
+    @property({ type: String }) rel = "";
+    @property({ type: String }) name = "";
+    @property({ type: String }) action = "";
+    @property({ type: String }) label = "";
+    @property({ type: String }) method: Method = "GET";
+    @property({ type: String }) target: Target = "_self";
+    @property({ type: String }) enctype: Enctype = "application/x-www-form-urlencoded";
+    @property({ type: Boolean, reflect: true }) novalidate = false;
```

Comment:
I believe when we have a boolean property that is more than 1 word, we make sure to add an attribute value that provides a camelcase variable name. ex. `no-validate`.

Reply:
Agreed

Reply:
@elenagiere change property name to more appropriate